bot/exts/moderation/silence.py

```diff
@@ -92,88 +102,221 @@ async def _async_init(self) -> None:
         """Set instance attributes once the guild is available and reschedule unsilences."""
         await self.bot.wait_until_guild_available()
 
-        guild = self.bot.get_guild(Guild.id)
+        guild = self.bot.get_guild(constants.Guild.id)
+
         self._everyone_role = guild.default_role
-        self._mod_alerts_channel = self.bot.get_channel(Channels.mod_alerts)
-        self.notifier = SilenceNotifier(self.bot.get_channel(Channels.mod_log))
+        self._verified_voice_role = guild.get_role(constants.Roles.voice_verified)
+        self._helper_role = guild.get_role(constants.Roles.helpers)
+
+        self._mod_alerts_channel = self.bot.get_channel(constants.Channels.mod_alerts)
+
+        self.notifier = SilenceNotifier(self.bot.get_channel(constants.Channels.mod_log))
         await self._reschedule()
 
+    async def send_message(
+        self,
+        message: str,
+        source_channel: TextChannel,
+        target_channel: TextOrVoiceChannel,
+        alert_target: bool = False
+    ) -> None:
+        """Helper function to send message confirmation to `source_channel`, and notification to `target_channel`."""
+        # Reply to invocation channel
+        source_reply = message
+        if source_channel != target_channel:
+            source_reply = source_reply.replace("current channel", target_channel.mention)
+        await source_channel.send(source_reply)
+
+        # Reply to target channel
+        if alert_target:
+            if isinstance(target_channel, VoiceChannel):
+                voice_chat = self.bot.get_channel(VOICE_CHANNELS.get(target_channel.id))
+                if voice_chat and source_channel != voice_chat:
+                    await voice_chat.send(message.replace("current channel", target_channel.mention))
+
+            elif source_channel != target_channel:
+                await target_channel.send(message)
+
+    async def _select_lock_channel(*args) -> TextOrVoiceChannel:
+        """Passes the channel to be silenced to the resource lock."""
+        channel = args[0].get("channel")
+        if channel is not None:
+            return channel
+
+        else:
+            return args[0].get("ctx").channel
+
     @commands.command(aliases=("hush",))
-    @lock_arg(LOCK_NAMESPACE, "ctx", attrgetter("channel"), raise_error=True)
-    async def silence(self, ctx: Context, duration: HushDurationConverter = 10) -> None:
+    @lock(LOCK_NAMESPACE, _select_lock_channel, raise_error=True)
+    async def silence(
+        self,
+        ctx: Context,
+        duration: HushDurationConverter = 10,
+        kick: bool = False,
+        *, channel: TextOrVoiceChannel = None
```

Comment:
Doesn't it make more sense to put the kick parameter after the channel? I understand you did it so you can consume all remaining text in case a voice channel is specified by a name containing spaces. However, I suspect most of the time the voice channel will be specified by its ID. The convenience of not having to give the default value for the `kick` parameter (which is _only_ relevant when `channel` is also specified) likely outweighs the convenience of not having to put quotes around channel names with spaces.

Reply:
I have had that exact same line of thought, but honestly, I don't feel too strongly about either case. I'll switch the order anyway because it's a lot more likely for the kick argument to not be passed.